Automattic/newspack-newsletters
# feat: add campaign monitor usage report

### All Submissions:






### Changes proposed in this Pull Request:



Adds usage report for Campaign Monitor

### How to test the changes in this Pull Request:

1. Make sure you have the Campaign Monitor credentials on Newsletters setting to an account that has some activity
2. `wp shell`
3.`$x = Newspack_Newsletters_Campaign_Monitor::instance()->get_usage_report();`
4. `$x->to_array();`

Make sure you get an output similar to:

```
=> array(8) {
  ["date"]=>
  string(10) "2023-11-14"
  ["emails_sent"]=>
  int(0)
  ["opens"]=>
  int(0)
  ["clicks"]=>
  int(0)
  ["unsubscribes"]=>
  int(0)
  ["subscribes"]=>
  int(2)
  ["total_contacts"]=>
  int(6)
  ["growth_rate"]=>
  float(0.3333333333333333)
}
```

Campaigns data are calculated with the difference between what happened between now and the last time the check was performed (this report is supposed to be fetched once a day)

If you have some Campaigns with activity, the first time you run the script emails_sent, clicks and open will all come with `0`.

If you want to test how the script would behave the next day, edit the values on the `'np_newsletters_campaign_data_' . $campaign_id` option. Set it to some values lower than what you have now and the report should output the difference.

### Other information:
